pkg/cluster/install.go

```diff
@@ -289,22 +289,34 @@ func setFieldCreatedByHive(createdByHive bool) database.OpenShiftClusterDocument
 
 func (m *manager) bootstrap() []steps.Step {
 	s := []steps.Step{
-		steps.AuthorizationRetryingAction(m.fpAuthorizer, m.validateResources),
+		// TODO: Uncomment this when Rajdeep's dynamic validation merges
+		//steps.AuthorizationRetryingAction(m.fpAuthorizer, m.validateResources),
 		steps.Action(m.ensurePreconfiguredNSG),
 		steps.Action(m.ensureACRToken),
 		steps.Action(m.ensureInfraID),
 		steps.Action(m.ensureSSHKey),
 		steps.Action(m.ensureStorageSuffix),
 		steps.Action(m.populateMTUSize),
-
 		steps.Action(m.createDNS),
 		steps.Action(m.createOIDC),
-		steps.Action(m.initializeClusterSPClients), // must run before clusterSPObjectID
+	}
 
-		// TODO: this relies on an authorizer that isn't exposed in the manager
-		// struct, so we'll rebuild the fpAuthorizer and use the error catching
-		// to advance
-		steps.AuthorizationRetryingAction(m.fpAuthorizer, m.clusterSPObjectID),
+	if m.doc.OpenShiftCluster.UsesWorkloadIdentity() {
+		s = append(s,
+			steps.Action(m.ensureClusterMsiCertificate),
+			steps.Action(m.initializeClusterMsiClients),
```

Comment:
This step will need to be performed before the dynamic validation, as there will be a need to populate the cliendid and objectid of the platform identities. Also via the MSI Certificate there will be a need to populate the objectid for cluster MSI. objectid will be needed to use the checkaccess api. 

The cluster MSI dynamic validation will contain:-
- Checking if the cluster MSI has sufficient permission to read platformidentities and read/update their corresponding federated credentials.

Also, the msi clients initialization will be required for the update flow, will that be covered in this story or somewhere else.

Reply:
Gotcha, I'll fix the order of the steps.

>Also, the msi clients initialization will be required for the update flow, will that be covered in this story or somewhere else?

I figured it could be handled as needed in another PR that actually uses the cluster MSI clients during cluster updates.

Reply:
Done!